lib/reservation_manager.rb

```diff
@@ -0,0 +1,134 @@
+require_relative "reservation"
+require "pry"
+
+module Hotel
+  class ReservationManager
+    attr_reader :reservation_array, :rooms, :blocked_reservation_array
+
+    def initialize
+      @booked_rooms = []
+      @available_rooms = []
+      @reservation_array = []
+      @blocked_rooms_array = []
+      @blocked_reservation_array = []
+      @rooms = ("1".."20").to_a
+    end
+
+    def make_reservation(start_date: Date.today.to_s, end_date: (Date.today + 1).to_s, room: "0", cost: 200, block_name: "no name")
+      new_reservation = Reservation.new(start_date: start_date, end_date: end_date, room: room, cost: cost, block_name: block_name)
+      check_res = view_available_rooms(start_date: new_reservation.start_date, end_date: new_reservation.end_date)
+      if !check_res.include?(new_reservation.room)
+        raise ArgumentError, "That room is not available, choose another room"
+      elsif new_reservation.block_name != "no name"
```

Comment:
I would suggest that `block_name` default to `nil` instead of `"no name"` just to keep someone from naming their block `"no name"`.